pkg/machine/convert.go

```diff
@@ -348,11 +348,10 @@ func GetAPIV2NodeCloudSpec(machineSpec clusterv1alpha1.MachineSpec) (*apiv1.Node
 				TemplateID: config.TemplateID.Value,
 				CPUs:       config.CPUs,
 				Memory:     int64(config.Memory),
-				DiskSize:   &[]int64{int64(config.DiskSize)}[0],
 			}
 
 			if config.DiskSize > 0 {
-				cloudSpec.Anexia.DiskSize = &[]int64{int64(config.DiskSize)}[0]
+				cloudSpec.Anexia.DiskSize = &[]int64{int64(config.DiskSize)}[config.DiskSize]
```

Comment:
Surely `k8s.io/utils/pointer` has a helper that prevents this .... creative approach to creating pointers.

Reply:
Same goes for `cloudSpec.Anexia.Disks[diskIndex].PerformanceType = &[]string{diskConfig.PerformanceType.Value}[0]` a bit further down.

Reply:
Took that from other code in this code base, actually learned it here ^^`

Reply:
Also using [config.DiskSize] as the index here is probably also unintended :grin:

Reply:
But can definitely change it, it's at least the only occurrence in this file - maybe the original ones I saw were in machine-controller even

Reply:
> Also using [config.DiskSize] as the index here is probably also unintended grin

oof right, thanks, too long day ... will deliver a fixed version first thing tomorrow - before I make more humiliating stuff today ^^'

Reply:
addressed both problems, thanks again :)

Reply:
It's now so much prettier, don't you agree? :D

Reply:
it is! but as said, I just oriented myself on existing code at some place :P

Reply:
However, it seems that it doesn't compile anymore... Weird.

Reply:
huh? I'll take a look

Reply:
Fix incoming